integrationtests/src/test/java/io/kroxylicious/proxy/filter/CompositePrefixingFixedClientIdFilter.java

```diff
@@ -50,4 +51,22 @@ public CompositePrefixingFixedClientIdFilterConfig(@JsonProperty("prefix") Strin
             this.clientId = clientId;
         }
     }
+
+    public static class Contributor implements FilterContributor {
+
+        @Override
+        public String getTypeName() {
+            return "CompositePrefixingFixedClientId";
+        }
+
+        @Override
+        public ConfigurationDefinition getConfigDefinition() {
+            return new ConfigurationDefinition(CompositePrefixingFixedClientIdFilterConfig.class, true);
+        }
+
+        @Override
+        public Filter getInstance(FilterConstructContext context) {
+            return new CompositePrefixingFixedClientIdFilter((CompositePrefixingFixedClientIdFilterConfig) context.getConfig());
```

Comment:
Every filter implementation having to cast is a bit sad, maybe we can add a little typing somewhere so the user doesn't have to cast.

Reply:
I've introduced a `ConfigurableFilterContributor` abstract class as a convenience to save the user casting. So this example becomes:

```java
public static class Contributor extends ConfigurableFilterContributor<CompositePrefixingFixedClientIdFilterConfig> {

        public Contributor() {
            super("CompositePrefixingFixedClientId", CompositePrefixingFixedClientIdFilterConfig.class, true);
        }

        @Override
        protected Filter getInstance(FilterConstructContext context, CompositePrefixingFixedClientIdFilterConfig config) {
            return new CompositePrefixingFixedClientIdFilter(config);
        }
    }
```